SEED-platform/seed
# Migrate BETTER API from V1 to V2

#### Any background context you want to provide?
The API service that the SEED platform uses to generate BETTER analysis has undergone a version change from V1 to V2. The V1 endpoints have been overwritten by the V2 endpoints breaking the existing functionality. It's worth noting that this is BETTER API V2, however, all endpoints still point to the `api/v1/` URL.

V2 makes changes to how Portfolio Analyses are processed (a Portfolio is a collection of Buildings). In V1 a portfolio analysis was initiated and a building_analysis was created for each building in the portfolio. In V2 instead of creating building_analyses, a portfolio analysis will generate a portfolio_building_analysis for each building in the portfolio.  

#### What's this PR do?
This PR migrates the BETTER pipeline functionality to the BETTER V2 endpoints. 
- new pathways to retrieve and store portfolio_building_analyses
- updates to variable naming 
- Add a new Org Settings page button to verify the current token.

#### How should this be manually tested?
To test the BETTER token verification button
- Go to organization settings > API Key > there is now a button to "Verify Token"
- The button text and color will change to indicate a valid or invalid token. 
- You must save changes to reverify the token

To test the BETTER analyses
- You must first have inventory that meets the better requirements. This includes meter data, property type, city, state, and gross floor area. Note that the property type "Office" is reliable.
- select one property and run a better analysis > ensure a successful run and the property now displays the analysis results
- select multiple properties and run a better analysis with "Run Portfolio Analysis" unchecked. Ensure a successful run and the selected properties show the analysis results
- select multiple properties and run a better analysis with "Run Portfolio Analysis" checked. Ensure a successful run and the selected properties show the analysis results

#### What are the relevant tickets?
#3874 

#### Screenshots (if appropriate)
![Screenshot 2023-02-28 at 3 06 06 PM](https://user-images.githubusercontent.com/58446472/221992788-c7f5e443-4979-411f-8e95-7d6382e76c31.png)


![Screenshot 2023-02-28 at 3 06 28 PM](https://user-images.githubusercontent.com/58446472/221992598-1da995ac-c695-475b-a146-2a40156cabbe.png)

![Screenshot 2023-02-28 at 3 07 02 PM](https://user-images.githubusercontent.com/58446472/221992596-3a200f2e-a856-49e3-82c0-995a8b67f168.png)

#### Known issues
BETTER V2 introduces an `enable_pvwatts` analysis configuration key. This key has been hardcoded to `false`. Integrating this key will require further design.

Of the 3 active BETTER servers, production and staging are functional, while development is missing endpoints.

Within the /buildingsync.py file of SEED, there are a number of property_types that are defined as accepted BETTER (and bsync) properties, however several of them will cause a BETTER analysis to fail.
- passing types: Office, Hotel, K-12 School, Multifamily Housing
- failing types: Hospital (General Medical & Surgical), Bank Branch, Courthouse, Data Center, Distribution Center, Financial Office, Non-Refrigerated Warehouse, Refrigerated Warehouse, Retail Store, Senior Care Community, Supermarket/Grocery Store, Other
